BLD: Explicitly set pandas

thequackdaddy opened the pull request:
Pandas appears to be entering version hell. That's breaking your build. 

Also, I removed the Linux python 3.7 i686 build as there's no pandas support for that, it seems. 

Lastly, I changed the tests to skip the slower tests. 

There still seems to be a failure on the Mac builds. I'll try to look into that. Its a little hard to replicate, however. The mac builds on the statsmodels repo work. Statsmodels Mac support is known to be temperamental, granted.

bashtage commented:
@thequackdaddy Can you run again when you get a chance.  Patches in master against these.

thequackdaddy commented:
Amended and force-pushed.

bashtage commented:
Closing in - travis is green now.

bashtage commented:
I think I've squished all of the random-ish failures (some of which were exactly random, some which seem to be numerical precision related, especially when there is a non-linear optimizer involved).

thequackdaddy commented:
Issue is here-ish 

https://travis-ci.org/MacPython/statsmodels-wheels/jobs/531389487

In essence, I think there are 2 dicts used to parametrize in /stats/tests/test_multi.py 

I found a comment in the pytest forums that you can't use dicts or sets to do that because they are unsorted. Trying to understand the issue more. 

https://github.com/pytest-dev/pytest-xdist/issues/149#issuecomment-304803645

thequackdaddy commented:
To be clear, the issue is with pytest-xdist, not randomly. I was just trying to figure out the cause. But I'm fairly confident in the "dicts aren't sorted, and that confused xdist". 

With that said, xdist seems slow on linux... you can do "-n 2", but it saves no time. It might actually add to the time.

bashtage commented:
To get gains from Linux, which gives you 1.5 vcpu, you have to set mkl num
threads, open Las num threads, etc to 1.

On Sun, May 12, 2019, 13:44 Peter Quackenbush <notifications@github.com>
wrote:

> To be clear, the issue is with pytest-xdist, not randomly. I was just
> trying to figure out the cause. But I'm fairly confident in the "dicts
> aren't sorted, and that confused xdist".
>
> With that said, xdist seems slow on linux... you can do "-n 2", but it
> saves no time. It might actually add to the time.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/MacPython/statsmodels-wheels/pull/2#issuecomment-491592488>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABKTSRI2NUISYXPFO74NURLPVAGKXANCNFSM4HLJWYTA>
> .
>

bashtage commented:
I'll change these to ordered dicts in master.

On Sun, May 12, 2019, 13:55 Kevin Sheppard <kevin.k.sheppard@gmail.com>
wrote:

> To get gains from Linux, which gives you 1.5 vcpu, you have to set mkl num
> threads, open Las num threads, etc to 1.
>
> On Sun, May 12, 2019, 13:44 Peter Quackenbush <notifications@github.com>
> wrote:
>
>> To be clear, the issue is with pytest-xdist, not randomly. I was just
>> trying to figure out the cause. But I'm fairly confident in the "dicts
>> aren't sorted, and that confused xdist".
>>
>> With that said, xdist seems slow on linux... you can do "-n 2", but it
>> saves no time. It might actually add to the time.
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/MacPython/statsmodels-wheels/pull/2#issuecomment-491592488>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ABKTSRI2NUISYXPFO74NURLPVAGKXANCNFSM4HLJWYTA>
>> .
>>
>

thequackdaddy commented:
Hold on. I am testing it first.

thequackdaddy commented:
Meant to add some finishing notes... 

1. I changed the code so xdist will use both cores on the Appveyor and travis Mac builds. Saves a bunch of time. 
2. Also added pytest-randomly which is not neccessary (I think) but consistent with the main repo. 
3. There are still random fails. Like the most recent appveyor build seems like it it just crashed randomly. I'm hoping that doesn't happen any more frequently 

So this is ready to go now or a can squash. Likely we can wait until the next statsmodels release.

thequackdaddy commented:
I also meant to say "happy to squash" if u want. Lotsa irrelevant experimenting commits in there.

bashtage commented:
You could probably kill xdist on Windows.  The last build had a worker crash.

thequackdaddy commented:
I amended and force-pushed without removing xdist, and it worked this time. I think that was just a fluke. I hadn't seen that particular failure before. Do you still want to remove xdist on Windows? It cuts the build time close to half.

bashtage commented:
I suppose since we are really trying to build wheels it doesn't matter that much about how quick it is, but it is more annoying to get a random process-crash failure. Maybe leave it in for now and we'll see what happens.  I've never seen that crash.

bashtage commented:
One final question, what consideration is setting the build deps on some of the builds (and not on others?)

thequackdaddy commented:
That's probably a better question for Matthew Brett.

But as a guess, there is a global numpy/scipy version in .travis.yml and appveyor.yml files. That is for a pretty old scipy/numpy. I think that version is not supported on newer pythons, so a more recent one is used there. Still, I would imagine that the versions should all be forward compatible for some time. 

The pandas dep is simply needed because pandas is used to test. However, pandas is unnecessary for the build.

bashtage commented:
I think there are som good reasons to try newer numpy in particular since
openblas has had its share of problems in the past.  But it would be good
to get some idea about what to use and why.

On Fri, 17 May 2019 at 15:40 Peter Quackenbush <notifications@github.com>
wrote:

> That's probably a better question for Matthew Brett.
>
> But as a guess, there is a global numpy/scipy version in .travis.yml and
> appveyor.yml files. That is for a pretty old scipy/numpy. I think that
> version is not supported on newer pythons, so a more recent one is used
> there. Still, I would imagine that the versions should all be forward
> compatible for some time.
>
> The pandas dep is simply needed because pandas is used to test. However,
> pandas is unnecessary for the build.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/MacPython/statsmodels-wheels/pull/2?email_source=notifications&email_token=ABKTSROO5OJ5C5G6AHF66ATPV27UFA5CNFSM4HLJWYTKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVU6DII#issuecomment-493478305>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABKTSRI4TJEGF5PVGZAHI2TPV27UFANCNFSM4HLJWYTA>
> .
>

bashtage commented:
If this passes then I'm going to tag the RC.

bashtage commented:
@thequackdaddy I have tagged the RC2. The sdist isn't on PyPI yet since I need some permissions from @josef-pkt .  Once this is up, can you build the rc2 wheels?

bashtage commented:
@thequackdaddy rc2 is now up on pypi.  Can you build the wheels?

thequackdaddy commented:
@matthew-brett If I understand correctly, you have to merge this. If there is a way we can do it and get the uploaded wheels, let me know.

matthew-brett commented:
@thequackdaddy - shall I add you as a collaborator, so you can merge?

thequackdaddy commented:
@matthew-brett Could you add me and @bashtage, please?

matthew-brett commented:
Invites sent.
